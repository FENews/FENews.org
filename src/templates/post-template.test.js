import React from 'react';
import renderer from 'react-test-renderer';
import PostTemplate from './post-template';

describe('PostTemplate', () => {
  const props = {
    data: {
      site: {
        siteMetadata: {
          title: 'test',
          subtitle: 'test'
        }
      },
      markdownRemark: {
        html: '<p>test</p>',
        fields: {
          tagSlugs: [
            '/test_0',
            '/test_1'
          ]
        },
        frontmatter: {
          date: '2016-09-01',
          description: 'test',
          title: 'test',
          tags: [
            'test_0',
            'test_1'
          ]
        }
      }
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<PostTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
