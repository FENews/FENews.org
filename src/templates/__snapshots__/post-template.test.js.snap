// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTemplate renders correctly 1`] = `
<div
  className="layout"
>
  <div
    className="post"
  >
    <Link
      className="post__home-button"
      to="/"
    >
      All Articles
    </Link>
    <div
      className="post__content"
    >
      <div
        className="content"
      >
        <h1
          className="content__title"
        >
          test
        </h1>
        <div
          className="content__body"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>test</p>",
            }
          }
        />
      </div>
    </div>
    <div
      className="post__footer"
    >
      <div
        className="meta"
      >
        <p
          className="meta__date"
        >
          Published 
          1 Sep 2016
        </p>
      </div>
      <div
        className="tags"
      >
        <ul
          className="tags__list"
        >
          <li
            className="tags__list-item"
          >
            <Link
              className="tags__list-item-link"
              to="/test_0"
            >
              test_0
            </Link>
          </li>
          <li
            className="tags__list-item"
          >
            <Link
              className="tags__list-item-link"
              to="/test_1"
            >
              test_1
            </Link>
          </li>
        </ul>
      </div>
      <StaticQuery
        render={[Function]}
      />
    </div>
    <div
      className="post__comments"
    >
      <StaticQuery
        render={[Function]}
      />
    </div>
  </div>
</div>
`;
